app/v3/controllers/RetrieveSelfEmploymentBsasController.scala

```diff
@@ -89,9 +89,17 @@ class RetrieveSelfEmploymentBsasController @Inject()(
 
   private def errorResult(errorWrapper: ErrorWrapper) =
     errorWrapper.error match {
-      case BadRequestError | NinoFormatError | CalculationIdFormatError | RuleTypeOfBusinessIncorrectError => BadRequest(Json.toJson(errorWrapper))
-      case NotFoundError                                                                                   => NotFound(Json.toJson(errorWrapper))
-      case InternalError                                                                                 => InternalServerError(Json.toJson(errorWrapper))
-      case _               => unhandledError(errorWrapper)
+      case _ if errorWrapper.containsAnyOf(
+        BadRequestError,
+        NinoFormatError,
+        TaxYearFormatError,
+        RuleTaxYearRangeInvalidError,
+        InvalidTaxYearParameterError,
+        CalculationIdFormatError,
+        RuleTypeOfBusinessIncorrectError) => BadRequest(Json.toJson(errorWrapper))
+      case NotFoundError                  => NotFound(Json.toJson(errorWrapper))
+      case RuleTaxYearNotSupportedError   => InternalServerError(Json.toJson(errorWrapper))
```

Comment:
`RuleTaxYearNotSupportedError` should be mapped to `BadRequest` not `InternalServerError`